### PR TITLE
Fix immutable support

### LINKED ISSUE #4473: Possible error in the last release
Please fix:

```
Failed to compile
./node_modules/redux-form/es/handleSubmit.js
Module not found: Can't resolve 'immutable' in '/var/www/faithmo-dev/node_modules/redux-form/es'
```

It works for `"redux-form": "8.2.0"` but fails for `"redux-form": "^8.2.0"` in my project.

### PR DESCRIPTION
**Pull request by esetnik**
Third time is the charm.

Fixes #4473

### CODE REVIEW

**Comment by renatoagds** [comment]
Sorry for the delay. I was just double checking if we don't have any critical bug.

**Comment by renatoagds** [comment]
Published in `[8.2.4](https://github.com/erikras/redux-form/releases/tag/v8.2.4)`

File: src/handleSubmit.js [hunk 1]
```diff
@@ -11,9 +12,24 @@ const mergeErrors = ({ asyncErrors, syncErrors }) =>
     ? asyncErrors.merge(syncErrors).toJS()
     : { ...asyncErrors, ...syncErrors }
 
+let isImmutableList
+try {
+  // ImmutableJS isList implementation if available
+  // eslint-disable-next-line import/no-extraneous-dependencies
+  const { List } = require('immutable')
```
Comment (iamandrewluca) [inline src/handleSubmit.js]: This creates a new issue. Now when building an app that uses `redux-form` it includes `immutable` in the bundle #4519

File: package.json [hunk 2]
```diff
@@ -141,6 +143,9 @@
     "react-redux": "^6.0.0 || ^7.0.0",
     "redux": "^3.0.0 || ^4.0.0"
   },
+  "optionalDependencies": {
+    "immutable": "*"
```
Comment (iamandrewluca) [inline package.json]: it seems this will still install the dependency

**Comment by iamandrewluca** [comment]
https://github.com/erikras/redux-form/pull/4474

**Pull request merged**